src/main/java/fr/openent/crre/controllers/EquipmentController.java

```diff
@@ -93,66 +103,97 @@ public void equipment(final HttpServerRequest request) {
         try {
             HashMap<String, ArrayList<String>> params = new HashMap<>();
-            getAllWithFilter(request, params);
+            getCatalog(params, false)
+                    .onSuccess(result -> Renders.renderJson(request, result))
+                    .onFailure(error -> Renders.renderError(request));
         } catch (ClassCastException e) {
             log.error("An error occurred casting campaign id", e);
         }
     }
 
-    private void getAllWithFilter(HttpServerRequest request, HashMap<String, ArrayList<String>> params) {
-        equipmentService.filterWord(params, null, event -> {
-            JsonArray ressources = event.right().getValue();
-            JsonArray filtres = new JsonArray();
-            JsonArray response = new JsonArray();
-            Set<String> disciplines_set = new HashSet<>();
-            Set<String> niveaux_set = new HashSet<>();
-            Set<String> classes_set = new HashSet<>();
-            Set<String> editeur_set = new HashSet<>();
-            Set<String> public_set = new HashSet<>();
-            Set<String> os_set = new HashSet<>();
-            Set<String> distributeurs_set = new HashSet<>();
-            for(int i = 0; i < ressources.size(); i++) {
-                JsonObject ressource = ressources.getJsonObject(i);
-                for(int j = 0; j < ressource.getJsonArray("disciplines").size(); j ++) {
-                    disciplines_set.add(ressource.getJsonArray("disciplines").getJsonObject(j).getString("libelle"));
-                }
-                for(int j = 0; j < ressource.getJsonArray("niveaux").size(); j ++) {
-                    niveaux_set.add(ressource.getJsonArray("niveaux").getJsonObject(j).getString("libelle"));
-                }
-                if(ressource.containsKey("classes")) {
-                    for(int j = 0; j < ressource.getJsonArray("classes").size(); j ++) {
-                        classes_set.add(ressource.getJsonArray("classes").getJsonObject(j).getString("libelle"));
+    @Get("/equipments/catalog/filters")
+    @ApiDoc("List of item's filters")
+    @SecuredAction(value = "", type = ActionType.RESOURCE)
+    @ResourceFilter(AccessRight.class)
+    public void listItemsFilters(final HttpServerRequest request) {
+        try {
+            HashMap<String, ArrayList<String>> params = new HashMap<>();
+            getCatalog(params, true)
+                    .onSuccess(result -> Renders.renderJson(request, result))
+                    .onFailure(error -> Renders.renderError(request));
+        } catch (ClassCastException e) {
+            log.error("An error occurred casting campaign id", e);
+        }
+    }
+
+    private Future<JsonObject> getCatalog(HashMap<String, ArrayList<String>> params, Boolean onlyFilter) {
+        Promise<JsonObject> promise = Promise.promise();
+        List<String> fields = onlyFilter ? Arrays.stream(ResourceFieldEnum.values()).map(ResourceFieldEnum::getValue).collect(Collectors.toList()) : null;
```

Comment:
Risque NPE, utilise la condition primitive boolean` Boolean.TRUE.equals` pour check comme il faut ta valeur 

```suggestion
            List<String> fields = Boolean.TRUE.equals(onlyFilter) ? Arrays.stream(ResourceFieldEnum.values()).map(ResourceFieldEnum::getValue).collect(Collectors.toList()) : null;

```